include/nlohmann/json.hpp

```diff
@@ -3525,12 +3665,24 @@ class basic_json
                 m_value.array->insert(m_value.array->end(),
                                       idx - m_value.array->size() + 1,
                                       basic_json());
+#if JSON_DIAGNOSTICS
+                for (std::size_t i = idx + 1; i < m_value.array->size(); ++i)
+                {
+                    m_value.array->operator[](i).m_parent = this;
+                }
+#endif
             }
 
+#if JSON_DIAGNOSTICS
+            reference result = m_value.array->operator[](idx);
+            result.m_parent = this;
```

Comment:
It can't, because the insert above only happens if the array needs to be expanded.

Reply:
The parent should be set only when the array is expanded. Existing elements already have the parent.

Reply:
Ah, I see what you're saying now, that this assignment isn't necessary because it was previously done when the array was expanded to be large enough to handle this index.

Reply:
In fact, only after fixing the move constructor, I could drop that part of the code :)